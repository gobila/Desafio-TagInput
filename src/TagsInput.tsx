import React, { useEffect } from 'react'
import PropTypes from 'prop-types'
import Chip from '@material-ui/core/Chip'
import TextField from '@material-ui/core/TextField'

export default function TagsInput(props) {
  const { onSelectTags, tags, ...other } = props
  const [selectedItem, setSelectedItem] = React.useState([])
  useEffect(() => {
    setSelectedItem(tags)
  }, [tags])

  const handleDelete = (item) => () => {
    const lala = 'jajaja'
    // const Liked = postLiked.some((elem) => elem.user === user.id);
    // const newList = name.filter((item) => item.id !== itemId);
    console.log('deletando tag(item):', lala)
    setSelectedItem([lala])
    return item
  }

  return (
    <TextField
      InputProps={{
        startAdornment: selectedItem.map((item) => (
          <Chip key={item} tabIndex={-1} label={item} onDelete={handleDelete(item)} />
        )),
      }}
      {...other}
    />
  )
}
TagsInput.defaultProps = {
  tags: [],
}

TagsInput.propTypes = {
  onSelectTags: PropTypes.func,
  tags: PropTypes.array,
}
