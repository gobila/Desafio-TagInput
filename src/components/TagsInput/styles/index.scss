.tagInput{
  &__tag{
    display: flex;
    flex-wrap: wrap;
    padding: 6px;
  }
  &__chip{
    background-color: #4fa9f1;
    color:#000;
    margin:2px;
  }
  &__button{
    margin-bottom:20px;
    :hover{
      opacity: 0.5;
    }
  }
}