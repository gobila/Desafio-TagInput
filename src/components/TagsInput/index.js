import { Button, Chip, TextField } from '@material-ui/core'
import PropTypes from 'prop-types'
import React, { useEffect, useState } from 'react'
import { withStyles } from '@material-ui/core/styles'
import './styles/index.scss'

export default function TagsInput({ tags, changeValue, deleting, errors, onClick }) {
  const [emailTags, setEmailTags] = useState([])
  const deleted = deleting

  useEffect(() => {
    setEmailTags(tags)
  }, [tags])

  // console.log(errors)
  const styles = {
    root: {
      background: 'black',
    },
    input: {
      color: 'red',
    },
  }
  return (
    <div>
      <TextField
        tags={emailTags}
        onBlur={changeValue}
        fullWidth
        variant="outlined"
        id="emailsTags"
        name="email"
        label="Email"
        placeholder="add e-mails"
        // inputRef={register}
        error={errors}
        helperText={errors ? errors : ' '}
        InputProps={{
          className: 'tagInput__tag',
          startAdornment: emailTags.map((item) => (
            <Chip
              className="tagInput__chip"
              style={{ backgroundColor: '#4fa9f1', color: '#FFF' }}
              key={item}
              tabIndex={-1}
              label={item}
              onDelete={() => deleted(item)}
              // color="primary"
            />
          )),
        }}
      />
      <Button
        variant="contained"
        className="tagInput__button"
        style={{ backgroundColor: '#009688', color: '#FFF', marginBottom: '20px' }}
        onClick={onClick}
      >
        Enviar
      </Button>
    </div>
  )
}
TagsInput.defaultProps = {
  tags: [],
}

TagsInput.propTypes = {
  tags: PropTypes.array,
  changeValue: PropTypes.func,
  deleting: PropTypes.func,
  errors: PropTypes.string,
  onClick: PropTypes.func,
}
