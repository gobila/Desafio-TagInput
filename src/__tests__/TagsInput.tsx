import React from 'react'
import TagsInput from '@/components/TagsInput'
import { cleanup, render, screen } from '@testing-library/react'
import user from '@testing-library/user-event'

describe('TagsInput Component', () => {
  afterEach(jest.clearAllMocks)
  afterEach(cleanup)

  it('cria o snapshot do component', () => {
    const container = render(<TagsInput />)
    expect(container.asFragment()).toMatchSnapshot()
  })

  it('deve renderizar as tags enviadas por atributos', () => {
    const emails = ['contato@rarolabs.com.br', 'nao-responda@rarolabs.com.br']

    const { debug } = render(<TagsInput tags={emails} />)

    debug()
  })

  it('deve ser capaz de preencher o campo', () => {
    const emails = ['contato@rarolabs.com.br', 'nao-responda@rarolabs.com.br']
    render(<TagsInput tags={emails} />)

    const inputName = screen.getByPlaceholderText(/add e\-mails/i)
    const newEmail = 'teste@teste.com'
    user.type(inputName, newEmail)
    expect(inputName).toHaveValue(newEmail)
  })

  it('deve renderizar tags quando preencher o input e pressionar enter', () => {
    // teste não implementado.
  })

  it('deve renderizar tags quando preencher o input e pressionar tab', () => {
    const emails = ['contato@rarolabs.com.br', 'nao-responda@rarolabs.com.br']
    const handleTag = jest.fn()
    const container = render(<TagsInput tags={emails} changeValue={handleTag} />)
    expect(container.asFragment()).toMatchSnapshot()

    const inputName = screen.getByPlaceholderText(/add e\-mails/i)
    const newEmail = 'teste@teste.com'
    user.type(inputName, newEmail)
    expect(inputName).toHaveValue(newEmail)
    expect(user.type(inputName, '{tab}')).toBe.call(handleTag)
  })

  it('deve deletar a útima tag criada ao pressionar o botão de backspace', async () => {})
  // teste não implementado.
})
