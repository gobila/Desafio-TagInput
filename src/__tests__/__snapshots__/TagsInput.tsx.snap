// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagsInput Component cria o snapshot do component 1`] = `
<DocumentFragment>
  <div>
    <div
      class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
      tags=""
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined"
        data-shrink="true"
        for="emailsTags"
      >
        Email
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart MuiOutlinedInput-adornedStart"
      >
        <fieldset
          aria-hidden="true"
          class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
          style="padding-left: 8px;"
        >
          <legend
            class="PrivateNotchedOutline-legend-2"
            style="width: 0px;"
          >
            <span>
              â€‹
            </span>
          </legend>
        </fieldset>
        <input
          aria-describedby="emailsTags-helper-text"
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart MuiOutlinedInput-inputAdornedStart"
          id="emailsTags"
          name="email"
          placeholder="add e-mails"
          type="text"
          value=""
        />
      </div>
      <p
        class="MuiFormHelperText-root MuiFormHelperText-contained"
        id="emailsTags-helper-text"
      >
         
      </p>
    </div>
  </div>
</DocumentFragment>
`;
